examples/md-flexible/src/domainDecomposition/DomainTools.cpp

```diff
@@ -72,4 +71,50 @@ void generateDecomposition(unsigned int subdomainCount, std::array<int, 3> &deco
     }
   }
 }
+
+int convertIdToIndex(const std::array<int, 3> &domainId, const std::array<int, 3> decomposition) {
+  int domainIndex = 0;
+
+  for (size_t i = 0; i < 3; ++i) {
+    domainIndex += getAccumulatedTail(i, decomposition) * domainId[i];
+  }
+
+  return domainIndex;
+}
+
+std::array<int, 3> convertIndexToId(int domainIndex, const std::array<int, 3> decomposition) {
+  std::array<int, 3> id;
+
+  for (size_t i = 0; i < 3; ++i) {
+    int accumulatedTail = getAccumulatedTail(i, decomposition);
+    ;
+    id[i] = domainIndex / accumulatedTail;
+    domainIndex -= accumulatedTail * id[i];
+  }
+
+  return id;
+}
+
+size_t getAccumulatedTail(const size_t index, const std::array<int, 3> decomposition) {
+  int accumulatedTail = 1;
+  if (index < decomposition.size() - 1) {
+    accumulatedTail =
+        std::accumulate(decomposition.begin() + index + 1, decomposition.end(), 1, std::multiplies<int>());
```

Comment:
```suggestion
        std::accumulate(decomposition.begin() + index + 1, decomposition.end(), 1, std::multiplies<>());
```
should actually work